compiler/circle-mpqsolver/src/bisection/VISQErrorApproximator.cpp

```diff
@@ -17,23 +17,45 @@
 #include "VISQErrorApproximator.h"
 
 #include <fstream>
+#include <json.h>
 
 using namespace mpqsolver::bisection;
 
 void VISQErrorApproximator::init(const std::string &visq_data_path)
 {
   // read file
   std::ifstream file(visq_data_path);
-  if (!init(file))
+  try
+  {
+    init(file);
+  }
+  catch (std::runtime_error &)
   {
     throw std::runtime_error("Invalid visq file " + visq_data_path);
   }
 }
 
-bool VISQErrorApproximator::init(std::istream &)
+void VISQErrorApproximator::init(std::istream &visq_stream)
 {
-  // TODO
-  return true;
+  Json::Reader reader;
+  Json::Value completeJsonData;
+  if (!reader.parse(visq_stream, completeJsonData))
+  {
+    throw std::runtime_error("Invalid visq stream");
```

Comment:
this message isn't used as catch in `init(const std::string &visq_data_path)
{` ignores this.